api/path_agreement.go

```diff
@@ -44,6 +49,32 @@ func FindAgreementsForOutput(db *bolt.DB) (map[string]map[string][]persistence.E
 	return wrap, nil
 }
 
+// Obscure the secret details for all secrets in the given proposal
+func obscureSecretDetails(p *persistence.EstablishedAgreement) error {
```

Comment:
And you can use the code from the Policy type in both the agent and the agbot, so you wont have to duplicate it here.